src/main/java/robot/subsystems/drivetrain/pure_pursuit/Path.java

```diff
@@ -235,19 +239,33 @@ public void generateFillPoint() {
     }
 
     /**
-     * @param weight_data   amount of data
-     * @param weight_smooth amount of smooth
-     * @param tolerance     the min change between points
-     * @return the new path with the way points
+     * Smooth the points in the path class.
+     * <p>
+     * The second of the five methods used in the path generation, needed for the pure pursuit.
+     * (Pure pursuit article, 'Path Generation' > 'Smoothing' , Page 5)
+     *
+     * @param weight_data   smooth constant. controls the proportional distance from the original point.
+     * @param weight_smooth smooth constant. controls the proportional distance from the midpoint.
+     * @param tolerance     the minimum change between points.
+     * @return the new path with the way points.
      * @author Paulo
      * @author Lior
      */
     public void generateSmoothing(double weight_data, double weight_smooth, double tolerance) {
+        /*
+         * For simplification and parallelism with the article with the pure pursuit article, we converted the Waypoint
+         * path into a matrix of doubles. the first dimension represents the index of the waypoint, the second dimension
+         * the first column (column [0]) holding the x values and the second column (column [1]) holding the Y values.
+         *
+         * this smoothing algorithm takes each point on the path, tries to move it towards the midpoint of the next and
+         * previous point, proportionally to the distance from the midpoint, minus the amount it already moved until now.
+         * the program ends when the amount of change passes the tolerance.
+         */
         Path newPathClass = this.copy();
         double[][] newPath = new double[this.length()][2];
-        double a = weight_data;
-        double b = weight_smooth;
-        for (int i = 0; i < this.length(); i++) {
+        double a = weight_data; //multiplied by the distance the point has already moved.
```

Comment:
Declaring here two new variables, which are identical to the parameters, is unnecessary. Either remove these variables (and don't forget to switch every occurrence of them with the parameters), or rename the parameters.